src/lib/libast/tests/string/strtonll.c

```diff
@@ -0,0 +1,30 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <string.h>
+
+#include "ast.h"
+#include "terror.h"
+
+tmain() {
+    UNUSED(argc);
+    UNUSED(argv);
+
+    int64_t actual_result;
+    char *error;
+
+    struct StringToIntegerTest tests[] =  {
+        { "500", 500 },
+        { "999", 999 },
+        { "-123", -123 },
```

Comment:
Add a `{ "0", 0 }` test. It would also be good to verify the non-decimal values are handled correctly; e.g., `{"0x10", 16}, {"010", 8}`.